dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/ConnectorArguments.java

```diff
@@ -663,31 +661,31 @@ public String getUser() {
     return getOptions().valueOf(optionUser);
   }
 
-  // -password has optional argument, and if not provied
-  // should be asked from command line
+  @Nonnull
+  public String getUserOrFail() {
+    String user = getOptions().valueOf(optionUser);
+    if (user == null) {
+      throw new MetadataDumperUsageException("Required username was not provided.");
+    }
+    return user;
+  }
+
   @CheckForNull
-  public String getPassword() {
-    if (!getOptions().has(optionPass)) {
+  public String getPasswordIfProvided() {
```

Comment:
I would suggest renaming to `getPasswordIfFlagProvided` and adding a javadoc explaining what the method does along with the reference to the other `getPassword*` method.

Reply:
Done, I renamed the method and added a javadoc